src/weights.jl

```diff
@@ -466,11 +466,11 @@ w = rand(n)
 mean(x, weights(w))
 ```
 """
-mean(A::AbstractArray{T}, w::AbstractWeights{W};
```

Comment:
Better keep `W<:Real` for consistency as we do that everywhere. We could probably remove it, but let's do that separately (for example when moving these to Statistics).

Reply:
We don't in `mean!`, which is why I removed it--for consistency. :upside_down_face: